dev-libs/libflatarray/libflatarray-0.2.0-r1.ebuild

```diff
@@ -0,0 +1,32 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake cuda
+
+SRC_URI="http://www.libgeodecomp.org/archive/${P}.tar.bz2"
+KEYWORDS="~amd64 ~ppc ~x86"
+
+DESCRIPTION="Struct of arrays library with object oriented interface for C++"
+HOMEPAGE="http://www.libgeodecomp.org/libflatarray.html"
+
+SLOT="0"
+LICENSE="Boost-1.0"
+IUSE="cuda doc"
+
+DEPEND="
+	doc? ( app-doc/doxygen )
```

Comment:
This doesn't seem to have any function in this ebuild. Will the output be any different if I enable or disable `doc` for this package?

I checked upstream, they seem to have a potential for `examples` IUSE though, 
https://github.com/STEllAR-GROUP/libflatarray/blob/master/CMakeLists.txt#L122
https://github.com/STEllAR-GROUP/libflatarray/tree/master/examples

Reply:
Actually `doc` isn't used and it's useless. There is no documentation. I'll remove it.
Indeed `examples` might be useful. However, the cmake install target doesn't install them. So, two options: Installing them from the ebuild or providing a cmake patch e.g. with a option `WITH_EXAMPLES`. What do you think?

Reply:
There seems to be a configuration option for it, that at least seems to **build** the examples. You can then just call `dodoc -r ${BUILD_DIR}/examples/`or similar inside `if use examples; then` construct. But I haven't delved into upstream build system, that's just how it usually works.

Reply:
Added the `examples` useflag.